discovery/db_discovery.go

```diff
@@ -75,6 +77,22 @@ func (dbo *DBOrchestratorPoolCache) GetOrchestrators(numOrchestrators int) ([]*n
 	}
 
 	pred := func(info *net.OrchestratorInfo) bool {
+		protoParams := info.TicketParams
```

Comment:
Since this package imports the server package and this exact same workflow of conversion between protobuf `net.TicketParams` and `pm.TicketParams` types happens there in two files as well (`broadcast.go` and `segment_rpc.go`) , can we avoid code duplication and move this into a helper? 

```
getPMTicketParams(orchInfo *net.OrchestratorInfo) *pm.TicketParams {
  if orchInfo.TicketParams != nil {
    return &pm.TicketParams{
	Recipient:         ethcommon.BytesToAddress(orchInfo.TicketParams.Recipient),
	FaceValue:         new(big.Int).SetBytes(orchInfo.TicketParams.FaceValue),
	WinProb:           new(big.Int).SetBytes(orchInfo.TicketParams.WinProb),
	RecipientRandHash: ethcommon.BytesToHash(orchInfo.TicketParams.RecipientRandHash),
	Seed:              new(big.Int).SetBytes(orchInfo.TicketParams.Seed),
    }
  }
  return nil
}
```

Reply:
Fixed:

- 8e6c3f5: Use a helper in `discovery/db_discovery.go`
- 7fcae8c: Use a helper in `server/segment_rpc.go`
- c6b010e: Use helper in `server/broadcast.go`

I ended up defining the same `pmTicketParams()` helper in both the `discovery` and `server` packages for now. I know this introduces a bit of duplication across packages, but I was a little hesitant about defining the helper in the `server` package and then importing it in the `discovery` package. While the `discovery` package does already depend on the `server` package, it felt a bit weird to import a function from the `server` package that uses the `pm` and `net` packages when those packages are already imported in the `discovery` package. Plus, we should probably avoid turning `server` into a package that is depended on for exporting these types of helpers (i.e. converting `net` types into other types). So, I ended up prioritizing minimizing the amount of stuff that needs to be imported at the expense of a bit of code duplication.

Reply:
>  it felt a bit weird to import a function from the server package that uses the pm and net packages when those packages are already imported in the discovery package. Plus, we should probably avoid turning server into a package that is depended on for exporting these types of helpers

Great point and I should have thought of that considering my packet structure research.